### PR TITLE
Hoist Data Management code from XPCS and HEDM

### LINKED ISSUE #872: data management setup
Additional support is needed to setup and configure the APS data management device.  New work for APS XPCS beamline will show the pattern.

Additions include:

- `DM_SETUP_FILE` in `iconfig.yml`
- `framework/dm_setup.py`
- example in `devices/data_management.py`

### LINKED ISSUE #912: Add support DM DAQ
The APS Data Management has DAQ process that watch a file directory (and its subdirectories) for new files or modifications to existing ones.  A DAQ will copy such changes from the watched directory to a destination directory.  In addition to watching, a DAQ can launch a DM workflow.

Add support for DM's DAQs.

### PR DESCRIPTION
**Pull request by prjemian**
- close #872
- close #912

**Note**: Since we cannot test this code on GitHub, it is expected that the code coverage report from the GitHub workflow will decrease with these additions.

### CODE REVIEW

**Comment by prjemian** [comment]
Workflow is failing at

```
>       from dm import ProcApiFactory
E       ModuleNotFoundError: No module named 'dm'
```

For Python 3.11, the workflow has installed

```
2024-03-14T21:51:40.3065013Z   aps-dm-api                     5.0.3         py37_1                   aps-anl-tag
```

which is for Python 3.7!  Can't be found by Py3.11.

Because of this: https://github.com/BCDA-APS/apstools/blob/a39e177509d4cdafb8e54e0e4c62b3cf6b572c57/environment.yml#L38

**Comment by prjemian** [comment]
[Anaconda](https://anaconda.org/apsu/aps-dm-api) reports latest version is 8.0.2

**Comment by prjemian** [comment]
@MDecarabas @rodolakis - This is ready for review

File: docs/source/examples/de_aps_data_management.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+The steps to process data: move acquired data to
+computation hosts (local or remote), run the computation applications, then move
+any results to a final destination.  A workflow is capable of other activities,
+such as communicating back to EPICS.
+
+More info [here](https://git.aps.anl.gov/DM/dm-docs/-/wikis/DM/HowTos/Getting-Started#workflows-and-processing-data)
+
+### upload
+
+In addition to automatic file copying with a DAQ, DM provides for single file
```
Comment (sveseli) [inline docs/source/examples/de_aps_data_management.md:57]: Just to clarify, you can "upload" the whole directories as well. The difference between the "file" and "directory" processing modes is that in the file mode we files are processed (transfer, catalog, invoke workflow, etc) one by one, while  in the directory mode there will be a single transfer of the whole directory, after which all files will be cataloged, etc. The advantage of the file mode is that you can keep track of progress and see/recover from errors easier, while the advantage of the directory mode is that it is more efficient, especially for large number of small files, where any new transfer connection is more expensive relative to the data transfer time.

Comment (prjemian) [reply docs/source/examples/de_aps_data_management.md:57]: Thanks!

**Review by MDecarabas — APPROVED** [review]
LGTM!

**Pull request merged**